src/UseCases/CancelMembershipApplication/CancelMembershipApplicationUseCase.php

```diff
@@ -27,7 +31,63 @@ public function __construct( MembershipApplicationAuthorizer $authorizer,
 	}
 
 	public function cancelApplication( CancellationRequest $request ): CancellationResponse {
-		// TODO
+		$application = $this->getApplicationById( $request->getApplicationId() );
+
+		if ( $application === null ) {
+			return $this->newFailureResponse( $request );
+		}
+
+		$application->cancel();
+
+		try {
+			$this->repository->storeApplication( $application );
+		}
+		catch ( StoreMembershipApplicationException $ex ) {
+			// TODO: log?
+			return $this->newFailureResponse( $request );
+		}
+
+		$this->sendConfirmationEmail( $application );
```

Comment:
Can you create a new ticket for that under "new stuff"? (AFAIK this is not needed for feature parity)